src/MongoDB/Driver/WriteResult.cpp

```diff
@@ -197,16 +182,13 @@ Object hippo_write_result_init(mongoc_write_result_t *write_result, bson_error_t
 	}
 
 	if (success == 0) {
-		if (
-			bson_empty0(&write_result->writeErrors) &&
-			bson_empty0(&write_result->writeConcernErrors)
-		) {
-			throw MongoDriver::Utils::throwExceptionFromBsonError(error);
-		} else {
-			auto bw_exception = MongoDriver::Utils::throwBulkWriteException(message);
+		if (error->domain == MONGOC_ERROR_COMMAND || error->domain == MONGOC_ERROR_WRITE_CONCERN) {
+			auto bw_exception = MongoDriver::Utils::throwBulkWriteException(error->message);
```

Comment:
How does this work with https://github.com/mongodb/mongo-hhvm-driver/pull/77? If the BulkWrite is empty, `success` is obviously 0 and the error domain will be `MONGOC_ERROR_COMMAND` with code `MONGOC_ERROR_COMMAND_INVALID_ARG`. Will we end up throwing a BulkWriteException with an empty WriteResult attached?

Or will that InvalidArgumentException be thrown before we even get to this point?